test/remote/gateways/remote_pay_trace_test.rb

```diff
@@ -0,0 +1,201 @@
+    @options = {
+      billing_address: {
+        address1: '8320 This Way Lane',
+        city: 'Placeville',
+        state: 'CA',
+        zip: '85284'
+      },
+      description: 'Store Purchase'
+    }
+  end
+
+  def test_acquire_token
+    response = @gateway.acquire_access_token
+    assert_not_nil response['access_token']
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+    assert_equal 'Your transaction was successfully approved.', response.message
+  end
+
+  def test_successful_purchase_with_customer_id
+    create = @gateway.store(@mastercard, @options)
+    customer_id = create.params['customer_id']
+    response = @gateway.purchase(500, customer_id, @options)
+    assert_success response
+    assert_equal 'Your transaction was successfully approved.', response.message
+  end
+
+  def test_successful_purchase_with_more_options
+    options = {
+      email: 'joe@example.com'
+    }
+
+    response = @gateway.purchase(200, @discover, options)
+    assert_success response
+    assert_equal 'Your transaction was successfully approved.', response.message
+  end
+
+  def test_failed_purchase
+    response = @gateway.purchase(29, @amex, @options)
+    assert_failure response
+    assert_equal false, response.success?
+  end
+
+  def test_successful_authorize_and_capture
+    auth = @gateway.authorize(300, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(auth.authorization, @options.merge(amount: 3.00))
+    assert_success capture
+    assert_equal 'Your transaction was successfully captured.', capture.message
+  end
+
+  def test_successful_authorize_with_customer_id
+    store = @gateway.store(@mastercard, @options)
+    assert_success store
+    customer_id = store.params['customer_id']
+
+    response = @gateway.authorize(200, customer_id, @options)
+    assert_success response
+    assert_equal 'Your transaction was successfully approved.', response.message
+  end
+
+  def test_failed_authorize
+    response = @gateway.authorize(29, @mastercard, @options)
+    assert_failure response
+    assert_equal 'Your transaction was not approved.', response.message
+  end
+
+  def test_partial_capture
+    auth = @gateway.authorize(500, @amex, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(auth.authorization, @options.merge(amount: 3.00))
+    assert_success capture
+  end
+
+  def test_failed_capture
+    response = @gateway.capture('')
+    assert_failure response
+    assert_equal 'One or more errors has occurred.', response.message
+  end
+
+  def test_successful_refund
+    purchase = @gateway.purchase(200, @credit_card, @options)
+    assert_success purchase
+    authorization = purchase.authorization
+
+    settle = @gateway.settle()
+    assert_success settle
+
+    refund = @gateway.refund(authorization, @options.merge(amount: 1.00))
```

Comment:
I am not sure if it matters, but it looks like this is also a partial refund?

Reply:
I'll adjust for clarity and proper test naming 😄